src/aggregates/mod.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::VecDeque;
+use pgx::*;
+use pgx::Internal;
+use serde::{Deserialize, Serialize};
+
+use crate::palloc::{Inner, InternalAsValue};
+
+mod prom_delta;
+mod prom_increase;
+mod prom_rate;
+mod vector_selector;
+
+pub type Milliseconds = i64;
+pub type Microseconds = i64;
+
+pub const STALE_NAN: u64 = 0x7ff0000000000002;
+pub const USECS_PER_SEC: i64 = 1_000_000;
+pub const USECS_PER_MS: i64 = 1_000;
+
+#[pg_extern()]
+pub fn prom_delta_final(state: Internal) -> Option<Vec<Option<f64>>> {
+    prom_delta_final_inner(unsafe { state.to_inner() })
+}
+pub fn prom_delta_final_inner(
+    state: Option<Inner<GapfillDeltaTransition>>,
+) -> Option<Vec<Option<f64>>> {
+    state.map(|mut s| s.to_vec())
+}
```

Comment:
why are these here and not in prom_delta.rs

Reply:
Good question. `prom_delta_final` is used as the `finalfunc` for all aggregates. Perhaps it would make more sense to rename it, but I don't want to break compatibility (yet).

Reply:
ok let's just put in a code-comment todo to rename in the future